SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/CrossServerRPCService.cpp

```diff
@@ -398,6 +398,31 @@ void CrossServerRPCService::UpdateSentRPCsACKs(Worker_EntityId SenderId, const C
 			CrossServer::SentRPCEntry* SentRPC = SenderState.Mailbox.Find(RPCKey);
 			if (SentRPC != nullptr)
 			{
+				// If the ACK result is 'TargetUnknown' then resend the RPC immediately,
+				// as a temporary measure to bypass race conditions. There is deliberately
+				// no time-out handling at present.
+				if (ACK.Result == static_cast<uint64>(CrossServer::Result::TargetUnknown))
+				{
+					const Worker_EntityId TargetEntityId = SentRPC->Target.Entity;
+
+					const EntityViewElement& Entity = ActorSubView.GetView()[SenderId];
+					Schema_ComponentData* SenderComponentData =
+						Entity.Components
+							.FindByPredicate(ComponentIdEquality{ SpatialConstants::CROSS_SERVER_SENDER_ENDPOINT_COMPONENT_ID })
+							->GetUnderlying();
+
+					CrossServerEndpoint SenderEndpoint(SenderComponentData);
+					check(SentRPC->SourceSlot < static_cast<uint32>(SenderEndpoint.ReliableRPCBuffer.RingBuffer.Num()));
```

Comment:
ensure to prevent crashing?
Dropping the RPC if we can't find it again is not a catastrophe (even though it would mean we have wrong invariants).

Reply:
Thanks, I'll take  a look.